mlflow/tracking/fluent.py

```diff
@@ -3396,26 +3404,50 @@ def _get_active_model_context() -> ActiveModelContext:
 
 def get_active_model_id() -> Optional[str]:
     """
-    Get the active model ID. If no active model is set with ``set_active_model()``, this will
-    try to get the model ID from the environment variable ``MLFLOW_ACTIVE_MODEL_ID``.
-    If neither is set, return None.
+    Get the active model ID. If no active model is set with ``set_active_model()``, the
+    default active model is set using model ID from the environment variable
+    ``_MLFLOW_ACTIVE_MODEL_ID``. If neither is set, return None.
+    Note that this function only get the active model ID from the current thread.
 
     Returns:
         The active model ID if set, otherwise None.
     """
     return _get_active_model_context().model_id
 
 
-def _reset_active_model_context() -> None:
+def _get_active_model_id_global() -> Optional[str]:
     """
-    Should be called only for testing purposes.
+    Get the active model ID from the global context by checking all threads.
+    This is useful when we need to get the active_model_id set by a different thread.
     """
-    _ACTIVE_MODEL_CONTEXT.set(ActiveModelContext())
-    _update_active_model_id_env_var(None)
+    # if the active model ID is set in the current thread, always use it
+    if model_id_in_current_thread := get_active_model_id():
+        _logger.info(f"Active model ID found in the current thread: {model_id_in_current_thread}")
+        return model_id_in_current_thread
+    model_ids = [
+        ctx.model_id
+        for ctx in _ACTIVE_MODEL_CONTEXT.get_all_thread_values().values()
+        if ctx.model_id is not None
+    ]
+    if model_ids:
+        if len(set(model_ids)) > 1:
+            _logger.warning(
+                "Failed to get one active model id from all threads, multiple active model IDs "
+                f"found: {set(model_ids)}."
+            )
+            return
+        return model_ids[0]
+    elif _MLFLOW_DEBUG.get():
+        _logger.warning("No active model ID found in any thread.")
 
 
-def _update_active_model_id_env_var(value):
-    if value is None:
-        MLFLOW_ACTIVE_MODEL_ID.unset()
-    else:
-        MLFLOW_ACTIVE_MODEL_ID.set(value)
+def unset_active_model() -> None:
```

Comment:
For naming, do we have any precedent for `unset`? What about `clear` as an alternative - any precedent for that?

Reply:
We don't have precedant for any of them 😓 Most of the cases: we have set_... APIs and we allow passing None to them. So a similar move would be allowing passing `model_id=None` to `set_active_model` API, WDYT?

Reply:
how about `reset`?

Reply:
Does `reset` seem to imply that we're setting it to something non-null?

Reply:
Do we want to go with `clear_active_model`?